File: src/python/caesar.py
def encrypt(text, key):
   
    UPPER = 'ABCDEFGHIJKLMNOPQRSTUVWXYZ'
    LOWER = 'abcdefghijklmnopqrstuvwxyz'
    result = ''
    for char in text:
        if char.isupper():
            result += UPPER[(UPPER.index(char) + key) % 26]
        elif char.islower():
            result += LOWER[(LOWER.index(char) + key) % 26]
        else:
            result += char
    return result

def decrypt(text, key):

    return encrypt(text, -key)

if __name__ == "__main__":
    original = "Hello, World! ABCxyz"
    key = 3
    encrypted = encrypt(original, key)
    decrypted = decrypt(encrypted, key)
    print(f"Original: {original}")
    print(f"Encrypted: {encrypted}")
    print(f"Decrypted: {decrypted}")

File: src/javascript/caesar.js
function encrypt(message, key) {
    let result = '';
    for (let char of message) {
        if (char >= 'A' && char <= 'Z') {
            result += String.fromCharCode(((char.charCodeAt(0) - 65 + key) % 26) + 65);
        } else if (char >= 'a' && char <= 'z') {
            result += String.fromCharCode(((char.charCodeAt(0) - 97 + key) % 26) + 97);
        } else {
            result += char;
        }
    }
    return result;
}

function decrypt(message, key) {
    return encrypt(message, 26 - (key % 26));
}
console.log('Encrypt "Hello World" with key 3:', encrypt('Hello World', 3));
console.log('Decrypt "Khoor Zruog" with key 3:', decrypt('Khoor Zruog', 3));